src/compiler/checker.ts

```diff
@@ -1995,21 +2002,7 @@ namespace ts {
                 }
             }
             else if (name.kind === SyntaxKind.QualifiedName || name.kind === SyntaxKind.PropertyAccessExpression) {
-                let left: EntityNameOrEntityNameExpression;
-
-                if (name.kind === SyntaxKind.QualifiedName) {
-                    left = name.left;
-                }
-                else if (name.kind === SyntaxKind.PropertyAccessExpression) {
-                    left = name.expression;
-                }
-                else {

```

Comment:
this else clause can never happen because on line 2004 we just checked that name.kind is QualifiedName or PropertyAccessExpression. So I deleted it.